server/src/shared/query/pagination.ts

```diff
@@ -0,0 +1,50 @@
+import { IDeferredJoinParams, IKeysetParams } from 'shared/types/pagination';
+import db from 'shared/database';
+
+export async function deferredJoinPagination({
+  tableName,
+  indexedColumnName,
+  selectAttributes,
+  limit,
+  orderByDirection,
+  pageNumber,
+}: IDeferredJoinParams) {
+  const offset = (pageNumber - 1) * limit;
+  const innerQueryTableName = `${indexedColumnName}s_page`;
+  const attributes = selectAttributes || '*';
+  return db
+    .with(innerQueryTableName, db =>
+      db
+        .select(indexedColumnName)
+        .from(tableName)
+        .orderBy(indexedColumnName)
+        .limit(limit)
+        .offset(offset),
```

Comment:
Is the comma here deliberately?

Reply:
It was because I forgot to set the prettifier rule for no trailing commas.